### PR TITLE
Enable ns sorting linter and clean+sort all namespaces

### PR DESCRIPTION
**Pull request by ilmotta**
### Summary

This is a straightforward PR, please don't be scared by its size.

It does just two boring things:

1. Enable the clj-kondo rule to **force namespaces to be sorted**. The linter is enabled at the `error` level, so unsorted namespaces should _almost_ never be merged again. See [docs](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#unsorted-required-namespaces)
2. Call `clojure-lsp clean-ns && make lint-fix`.

**I changed absolutely nothing else in the code.**

### Why?

- Some devs would like to see this idea moving forward, like @smohamedjavid, @ulisesmac, @J-Son89 and I, but I guess others would appreciate 🙏🏼 
- We are not following the Clojure Style Guide, see [section for sorting requirements and imports](https://guide.clojure.style/#sort-requirements-and-imports).
- In PRs, it's common to see developers asking for namespaces to be sorted, but this can be automated.
- Sorting namespaces tend to reduce diffs because developers won't be changing ns requirements according to their own preferences.

### Merge strategy

- Wait for enough approvals, obviously ;)
- Rebase and quickly merge into `develop` when most of the contributors are sleeping 😴, that is, outside working hours between UTC +1 and UTC +5:30.

#### Areas that may be impacted

None.

status: ready

### CODE REVIEW

File: Makefile
Comment (ilmotta) [inline Makefile]: There is a discrepancy in how `zprint` formats namespaces and `clojure-lsp clean-ns`. That's why running `make lint-fix` won't fix sorting namespaces and that's why I kept this make target separate. @ulisesmac recently mentioned this formatting discrepancy here https://github.com/status-im/status-mobile/pull/17496#discussion_r1355844780

In VS Code, Vim and Emacs I know it's common and simple to clean up namespaces, either automatically after every save or by a keyboard shortcut, so hopefully this means not having `make lint-fix` doing the sorting will be fine. We could always enable the linter for unsorted namespaces at the warning level too, but then the code would still be merged with unsorted namespaces every now and then, so not much to gain I'd say.

**Review by yqrashawn — APPROVED** [review]

**Review by briansztamfater — APPROVED** [review]
LGTM 😃

**Review by flexsurfer — APPROVED** [review]

**Review by ibrkhalil — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]
A long-due task. 🚀 🚀

**Review by J-Son89 — APPROVED** [review]

**Comment by churik** [comment]
Tests passed, good to go if nothing affected from your POV :) 

thank you for following guidelines!

**Review by ulisesmac — APPROVED** [review]

**Comment by flexsurfer** [comment]
pls don't merge it yet. im moving quo. it will be a pain to resolve conflicts for me

**Comment by ilmotta** [comment]
> pls don't merge it yet. im moving quo. it will be a pain to resolve conflicts for me

Absolutely @flexsurfer 👍🏼

**Comment by flexsurfer** [comment]
done :) no blockers

File: src/status_im2/core.cljs
Comment (ilmotta) [inline src/status_im2/core.cljs:3]: FYI reviewers @flexsurfer, @yqrashawn, @briansztamfater, @smohamedjavid, @J-Son89, @ibrkhalil, @ulisesmac, also @churik 

I just spent more than 2h to identify the culprit for the translations not being loaded in their entirety. The problem is that `i18n-resources` must be loaded before other namespaces in the entrypoint namespace `status-im2.core`. `i18n-resources` is quite the stateful namespace as it preloads a bunch of things at "load time", so it makes sense the problem was there.

So if any of you would like to automate cleaning namespaces on every file save using `clojure-lsp`, be aware `status-im2.core` should not be cleaned by it.

**Pull request merged**
